tests/test_all

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+
+# Runs pre-commit, then all the scripts in this dir, and then builds documentaiton locally
+
+# Get the list of script files in the directory
+script_files=("$(dirname "$(realpath "$0")")"/*)
+num_files=$((${#script_files[@]} + 1))
+error_scripts=()
+
+echo "Found $num_files tests"
+
+echo "----- Running test 1 of $num_files: pre-commit"
+pre-commit run --all-files
+
+# Iterate over script files in the directory
+for script_file in "${script_files[@]}"; do
+
+  # Exclude the current script and non-executable files from execution
+  if [ "$(basename "$script_file")" != "test_all" ]; then
+    if [ -x "$script_file" ]; then
+      # Update progress
+      echo "----- Running test $((i + 1)) of $num_files: $script_file"
+
+      # Execute the script and capture errors if there are any
+      if output=$("$script_file" 2>&1); then
+        echo ":-) $script_file ran successfully."
+      else
+        echo "!!! $script_file encountered an error."
+        error_scripts+=("$script_file")
+        echo "$output" >&2
+      fi
+    else
+      # Notify non-executable files
+      echo "!!! File is not executable so was not run: $script_file."
+      echo
+    fi
+  fi
+done
+echo "----- Running test $num_files/$num_file: Building documentation locally"
```

Comment:
There is no `$num_file`, should be `$num_files`.

Reply:
good catch - didn't come up for me b/c it was in my env from before :-(

Reply:
fixed